Modules/IO/JPEG/src/itkJPEGImageIO.cxx

```diff
@@ -427,31 +439,19 @@ JPEGImageIO::WriteSlice(std::string & fileName, const void * const buffer)
   // struct jpeg_destination_mgr compressionDestination;
   jpeg_stdio_dest(&cinfo, fp);
 
-  // set the information about image
-  const SizeValueType width = m_Dimensions[0];
-  const SizeValueType height = m_Dimensions[1];
-
   // The JPEG standard only supports images up to 64K*64K due to 16-bit fields
   // in SOF markers.
   cinfo.image_width = width;
   cinfo.image_height = height;
-  if (cinfo.image_width > 65536 || cinfo.image_height > 65536)
```

Comment:
This is wrong. JPEG defines JPEG_MAX_DIMENSION in jmorecfg.h
`#define JPEG_MAX_DIMENSION  65500L  /* a tad under 64K to prevent overflows */`
External/system library may have modified jmorecfg.h too.